src/software/ai/hl/stp/action/action.h

```diff
@@ -39,6 +39,11 @@ class Action
      */
     bool done() const;
 
+    /**
+     * runs the coroutine from the start, effectively restarting the action
+     */
+    void restart();
```

Comment:
```suggestion
    void restart(void);
```
better to be explicit

Reply:
counter nit: not saying `void` is also pretty explicit, and is consistent with 99% of the codebase. No preference here since it's the same and doesn't impact readability, but this should be in the style guide if it's something you want to consistently enforce.

Reply:
https://stackoverflow.com/questions/51032/is-there-a-difference-between-foovoid-and-foo-in-c-or-c

There's a distinction in C  so that's why I'd prefer an explicit void but I agree it's not necessary in this case since it's C++

Reply:
Ah TIL: https://softwareengineering.stackexchange.com/questions/286490/what-is-the-difference-between-function-and-functionvoid/286494

We should consider adding to our firmware / C style guide if it's not there already.

Reply:
https://github.com/UBC-Thunderbots/Software/pull/1258

Reply:
Since theres no preference for c++ no changes were made from this